app/models/reviewer_list.rb

```diff
@@ -22,10 +22,29 @@ def each(&block)
       .each(&block)
   end
 
-  def choose_reviewer(exclude_list: [])
-    each
-      .reject { |r| exclude_list.include?(r.login) }
-      .sample
+  def choose_reviewer(pull_request, exclude_list: [])
+    filtered_reviewers = each
+                           .reject { |r| exclude_list.include?(r.login) }
+    # Average the load from last 7 days. Irrespective of the whether it is already reviewed or pending
+    reviewers_load = Reviewer.where("login IN (?) AND created_at > ?", filtered_reviewers.map(&:login), 1.week.ago).group(:login).count
```

Comment:
Since this is a reasonably complex query, let's extract it to a query object (`app/queries`) and write tests for the query specifically.

Some other points on the query itself:

I prefer to avoid the `?` style placeholders and use named placeholders instead.

ActiveRecord will generate the `IN` fragment automatically when given an Array value, so I would prefer to split out the two parts of this query into two `where` calls:

```suggestion
    reviewers_load = Reviewer.where(login: filtered_reviewers.map(&:login)).where("created_at > :one_week_ago", one_week_ago: 1.week.ago).group(:login).count
```